drivers/flash/flash_simulator.c

```diff
@@ -135,8 +135,9 @@ static int flash_sim_read(struct device *dev, const off_t offset, void *data,
 		return -EINVAL;
 	}
 
-	if ((offset % FLASH_SIMULATOR_PROG_UNIT) ||
-	    (len % FLASH_SIMULATOR_PROG_UNIT)) {
+	if (!CONFIG_FLASH_SIMULATOR_UNALIGNED_READ &&
```

Comment:
Consider moving CONFIG_FLASH_SIMULATOR_UNALIGNED_READ to the outer if statement. This way, the feature logic is more pronounced and easier to follow.
```suggestion
	if (!CONFIG_FLASH_SIMULATOR_UNALIGNED_READ) {
	(...)
	}
```

Reply:
I have merged your proposal and @nvlsianpu .